src/AppInstallerRepositoryCore/CompositeSource.cpp

```diff
@@ -467,7 +488,32 @@ namespace AppInstaller::Repository
                     // Search sources and add to result
                     for (const auto& source : m_availableSources)
                     {
-                        SearchResult availableResult = source->Search(systemReferenceSearch);
+                        // Do not attempt to correlate local packages against this source
+                        if (source->GetDetails().DoNotCorrelateAgainst)
+                        {
+                            continue;
+                        }
+
+                        SearchResult availableResult;
+
+                        try
+                        {
+                            availableResult = source->Search(systemReferenceSearch);
+                        }
+                        catch (...)
+                        {
+                            if (result.AddFailureIfSourceNotPresent({ source, std::current_exception() }))
+                            {
+                                LOG_CAUGHT_EXCEPTION();
+                                AICLI_LOG(Repo, Warning, << "Failed to search source for correlation: " << source->GetDetails().Name);
+                            }
+                        }
+
+                        // Move failures into the single result
+                        for (SearchResult::Failure& failure : availableResult.Failures)

```

Comment:
>availableResult.Failures [](http://example.com/codeflow?start=62&length=24)

I think this is from a single source(not composite), curious in what scenario will this have Failures entries?

Reply:
Ah, I saw this carries the OpenSource failure.

Reply:
I had originally done that, but this pattern made it cleaner to show the suggestions on a failed `install` command.